### ISSUE TITLE
Reattach not working when launching a program for debugging using ptvsd.enable_attach

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/ptvsd#516]
* Create a python program and enable remote debugig using `ptvsd.enable_attach`
```python
import ptvsd
import time

ptvsd.enable_attach()

for i in range(1000):
    time.sleep(1)
    print(i)
```
* Attach, then detach
* Attaching once again fails.

**Comment by karthiknadig** [comment]
Dupe of #472

**Comment by DonJayamanne** [comment]
@karthiknadig
Lol, you really need to provide more info on the issues you create, to prevent such dups. Cuz I thought that issue was for attaching programs launched using `python -m ptvsd ...`

**Comment by DonJayamanne** [comment]
Note I'm not getting any warning messages when disconnecting in VSC. Hence not sure they are the same.

**Comment by karthiknadig** [comment]
@DonJayamanne I had put `enable_attach` in the title, so I am not sure how it could be confused with `-m ptvsd ...`. I agree I should have provided the repro code. I didn't feel it was necessary at the time given this issue occurs with basic `enable_attach`. 

When I created the issue, the #398 was still open, it could have been a side effect of that issue.

**Comment by DonJayamanne** [comment]
My bad didn't read the title carefully enough.

**Comment by ericsnowcurrently** [comment]
This appears to be fixed with #527.
